app/Jobs/GenerateSocialShareImage.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace App\Jobs;
+
+use App\Models\Article;
+use Intervention\Image\ImageManager;
+
+final class GenerateSocialShareImage
+{
+    private $article;
+
+    const TEXT_X_POSITION = 50;
+
+    const TEXT_Y_POSITION = 100;
+
+    const TEXT_COLOUR = '#161e2e';
+
+    const FONT = 'inter.ttf';
+
+    const FONT_SIZE = 50;
+
+    const CHARACTERS_PER_LINE = 30;
+
+    const TEMPLATE = 'social-share-template.png';
+
+    const CACHE_LIFETIME = 43200;
+
+    public function __construct(Article $article)
+    {
+        $this->article = $article;
+    }
+
+    public function handle(ImageManager $image)
+    {
+        $text = wordwrap($this->article->title(), self::CHARACTERS_PER_LINE);
+
+        return $image->cache(function ($image) use ($text) {
+            $image->make(storage_path('images/' . self::TEMPLATE))
```

Comment:
Am I reading this right that there can only be one social image for one single article?

Reply:
Also: won't we get flooded disk space after a while?

Reply:
> Am I reading this right that there can only be one social image for one single article?

Right now yes. Did you have something else in mind?